applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1122,6 +1317,67 @@ namespace Kratos
     //************************************************************************************
     //************************************************************************************
 
+    void BaseSolidElement::CalculateAndAddKm(
```

Comment:
In that case i think I will optimize the product, I think that the material matrix is not a mistery